### DISCUSSION TITLE
Should AlertDialog have `onInteractionOutside`?

### DISCUSSION THREAD

**Discussion by franciscohanna92** [radix-ui/primitives#1777] [Help] [answered]
Hello!

I was wondering if it would make sense to have an `onInteractionOutside` callback on AlertDialog's Content element just like the one in Dialog's Content. I assume this is a design decision and in case it doesn't make sense I would like the reasoning behind it.

Regards.

**Comment by benoitgrelard** [comment, marked as answer]
It doesn't currently have `onInteractOutside` because it doesn't have `onPointerDownOutside`.
`onInteractOutside` is a combination of both `onPointerDownOutside` and `onEscapeKeydown`.

Currently `AlertDialog` only has `onEscapeKeydown` which arguably, perhaps we should remove.
The thinking behind this is that `AlertDialog` is for critical use input and so it should require an actual action from the user via one of the buttons, rather than soft dismissal.
